lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -293,26 +312,42 @@ combineChainData =
 -- hand-written Sqlite query.
 readDBPoolData
     :: DBLayer IO
-    -> IO (Map PoolId PoolDBMetrics)
-readDBPoolData DBLayer{..} = atomically $ do
+    -> IO (Map PoolId PoolDbData)
+readDBPoolData db@DBLayer{..} = atomically $ do
     pools <- listRegisteredPools
-    registrations <- mapM readPoolRegistration pools
+    registrationStatuses <- mapM (liftIO . readPoolRegistrationStatus db) pools
     let certMap = Map.fromList
-            [(poolId, cert) | (poolId, Just cert) <- zip pools registrations]
+            [ (poolId, certs)
+            | (poolId, Just certs) <- zip pools
+                (certificatesFromRegistrationStatus <$> registrationStatuses)
+            ]
     prodMap <- readTotalProduction
     metaMap <- readPoolMetadata
     return $ Map.map (lookupMetaIn metaMap) (combineChainData certMap prodMap)
   where
+    certificatesFromRegistrationStatus
+        :: PoolRegistrationStatus
+        -> Maybe (PoolRegistrationCertificate, Maybe PoolRetirementCertificate)
+    certificatesFromRegistrationStatus = \case
```

Comment:
> If PoolRegistrationStatus were replaced with the double Maybe then this function would not be needed.

That's true. But IMO this would reduce the readability of the following functions, both of which use the `PoolRegistrationStatus` type.

* `determinePoolRegistrationStatus`
* `readPoolRegistrationStatus`

See my comment here for a justification: https://github.com/input-output-hk/cardano-wallet/pull/1847#discussion_r451304532